src/dotnet/CarbonAware.WebApi/Controllers/SciScoreController.cs

```diff
@@ -56,39 +63,48 @@ public async Task<IActionResult> CreateAsync(SciScoreInput input)
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     public async Task<IActionResult> GetCarbonIntensityAsync(SciScoreInput input)
     {
-        _logger.LogInformation(" calling to aggregator to ");
-        if (input.Location == null)
-        {
-            var error = new CarbonAwareWebApiError() { Message = "location is required" };
-            return BadRequest(error);
-        }
-
-        if (String.IsNullOrEmpty(input.TimeInterval))
-        {
-            var error = new CarbonAwareWebApiError() { Message = "timeInterval is required" };
-            return BadRequest(error);
-        }
-
-        try
+        using (var activity = _activitySource.StartActivity(nameof(SciScoreController)))
         {
-            var location = this.GetLocation(input.Location);
-            var carbonIntensity = await _aggregator.CalculateAverageCarbonIntensityAsync(location, input.TimeInterval);
-            SciScore score = new SciScore
+            _logger.LogDebug("calling to aggregator to calculate the average carbon intensity with input: {input}", input);
+            
+            // check that there is some location passed in
+            if (input.Location == null)
             {
-                MarginalCarbonEmissionsValue = carbonIntensity,
-            };
+                var error = new CarbonAwareWebApiError() { Message = "Location is required" };
+                _logger.LogError("get carbon intensity failed with error: {error}", error);
+                return BadRequest(error);
+            }
 
-            return Ok(score);
-        }
-        catch (Exception ex)
-        {
-            _logger.LogError("Exception occured during marginal calculation execution", ex);
-            var error = new CarbonAwareWebApiError() { Message = ex.ToString() };
-            return BadRequest(error);
+            // check that there is a time interval passed in
+            if (String.IsNullOrEmpty(input.TimeInterval))
+            {
+                var error = new CarbonAwareWebApiError() { Message = "TimeInterval is required" };
+                _logger.LogError("get carbon intensity failed with error: {error}", error);
+                return BadRequest(error);
+            }
+            try
+            {
+                var carbonIntensity = await _aggregator.CalculateAverageCarbonIntensityAsync(GetLocation(input.Location), input.TimeInterval);
+
+                SciScore score = new SciScore
+                {
+                    MarginalCarbonIntensityValue = carbonIntensity,
+                };
+                _logger.LogDebug("calculated marginal carbon intensity: {score}", score);
+                return Ok(score);
+            }
+            // Catch ArgumentException for invalid inputs and broadly 3rd Party dependency exceptions
+            catch (Exception ex)
```

Comment:
I would propose two levels of catching and throwing here - one for the type of exception that you expect could happen (i.e. the ArgumentException type you note in your comment) and another "open" exception catcher to denote critical failures that you haven't expected

Reply:
valid, will be addressed in future PR in which all error types will be handled outside the controller